src/LinQuadOptInterface.jl

```diff
@@ -110,55 +110,71 @@ This function updates all of the references in `m`
 after we have deleted row `row` in the affine constraint matrix.
 """
 function shift_references_after_delete_affine!(m, row)
-    for scalar_affine in [
-            cmap(m).less_than,
-            cmap(m).greater_than,
-            cmap(m).equal_to,
-            cmap(m).interval
-        ]
-        for (key, val) in scalar_affine
-            if val > row
-                scalar_affine[key] -= 1
-            end
-        end
-    end
 
-    for vector_affine in [
-            cmap(m).nonnegatives,
-            cmap(m).nonpositives,
-            cmap(m).zeros,
-            cmap(m).vv_nonnegatives,
-            cmap(m).vv_nonpositives,
-            cmap(m).vv_zeros
-        ]
-        for (key, vals) in vector_affine
-            for (i, val) in enumerate(vals)
-                if val > row
-                    vector_affine[key][i] -= 1
-                end
-            end
+    cmap_m = cmap(m)
+    _shift_references_after_delete_scalar!(cmap_m.less_than, row)
+    _shift_references_after_delete_scalar!(cmap_m.greater_than, row)
+    _shift_references_after_delete_scalar!(cmap_m.equal_to, row)
+    _shift_references_after_delete_scalar!(cmap_m.interval, row)
+
+
+    _shift_references_after_delete_vector!(cmap_m.nonnegatives, row)
+    _shift_references_after_delete_vector!(cmap_m.nonpositives, row)
+    _shift_references_after_delete_vector!(cmap_m.zeros, row)
+    _shift_references_after_delete_vector!(cmap_m.vv_nonnegatives, row)
+    _shift_references_after_delete_vector!(cmap_m.vv_nonpositives, row)
+    _shift_references_after_delete_vector!(cmap_m.vv_zeros, row)
+
+end
+
+# The following two functions depend on the implimentation of Dict having the field vals
```

Comment:
s/implimentation/implementation

Reply:
No the changing the value operation is 2x slower but at that point, we are talking micro seconds.   It made no impact to the benchmarks so I changed to for loops.

Reply:
That makes much more sense. Can you exclude the partial start test as well please so that the tests pass? See https://github.com/JuliaOpt/Clp.jl/pull/54/commits/a60f8fe13b53847ff53ccc103d8f5567d9fb7682